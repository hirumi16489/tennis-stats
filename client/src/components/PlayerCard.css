

.card {
  background-color: #fff;
  max-width: 360px;
  display: flex;
  flex-direction: column;
  overflow: hidden;
  border-radius: 2rem;
  box-shadow: 0px 1rem 1.5rem rgba(0,0,0,0.5);
}
.card .banner {
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  height: 11rem;
  display: flex;
  align-items: flex-end;
  justify-content: center;
  box-sizing: border-box;
}
.card .banner img {
  background-color: #fff;
  width: 6rem;
  height: 8rem;
  box-shadow: 0 0.5rem 1rem rgba(0,0,0,0.3);
  border-radius: 50%;
  transform: translateY(50%);
  transition: transform 200ms cubic-bezier(0.18, 0.89, 0.32, 1.28);
}
.card .menu {
  width: 100%;
  height: 5.5rem;
  padding: 1rem;
  display: flex;
  align-items: flex-start;
  justify-content: flex-end;
  position: relative;
  box-sizing: border-box;
}
.card h2.name {
  text-align: center;
  padding: 0 2rem 0.5rem;
  margin: 0;
}
.card .title {
  color: #a0a0a0;
  font-size: 0.85rem;
  text-align: center;
  padding: 0 2rem 1.2rem;
}
.card .info-wrapper {
  padding: 0 2rem 1.2rem;
  display: flex;
  flex-direction: column;
  order: 99;
}
.card .info-wrapper .info {
  padding: 0 0 1rem;
  display: flex;
}
.card .info-wrapper .info.single {
  justify-content: center;
}
.card .info-wrapper .info h2 {
  text-align: center;
  width: 50%;
  margin: 0;
  box-sizing: border-box;
}
.card .info-wrapper .info h2 div {
  text-decoration: none;
  padding: 0.8rem;
  display: flex;
  flex-direction: column;
  border-radius: 0.8rem;
  transition: background-color 100ms ease-in-out;
}
.card .info-wrapper .info h2 div span {
  color: #5651e5;
  font-weight: bold;
  transform-origin: bottom;
}
.card .info-wrapper .info h2 div small {
  color: #afafaf;
  font-size: 0.85rem;
  font-weight: normal;
}

.card .desc {
  text-align: justify;
  padding: 0 2rem 2.5rem;
  order: 100;
}
