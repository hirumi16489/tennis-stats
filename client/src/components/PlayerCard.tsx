import React from 'react';
import './PlayerCard.css';
import { Player } from '../features/player/playerSlice';

type PlayerCardProps = {
  player: Player;
};

const PlayerCard = ({ player }: PlayerCardProps) => {
  return (
    <div className="card">
      <div
        className="banner"
        style={{ backgroundImage: `url(${player.country.picture})` }}
      >
        <img src={player.picture} alt="Avatar" />
      </div>
      <div>
        <div>
          <span></span>
          <span></span>
          <span></span>
        </div>
      </div>
      <h2 className="name">
        {player.firstname} {player.lastname}
      </h2>
      <div className="title">{player.shortname}</div>
      <div className="info-wrapper">
        <div className="info">
          <h2>
            <div>
              <span>{player.data.rank}</span>
              <small>Rank</small>
            </div>
          </h2>
          <h2>
            <div>
              <span>{player.data.points}</span>
              <small>Points</small>
            </div>
          </h2>
        </div>
        <div className="info">
          <h2>
            <div>
              <span>{player.data.weight / 1000}</span>
              <small>Weight</small>
            </div>
          </h2>
          <h2>
            <div>
              <span>{player.data.height / 100}</span>
              <small>Height</small>
            </div>
          </h2>
        </div>
        <div className="info single">
          <h2>
            <div>
              <span>{player.data.age}</span>
              <small>Age</small>
            </div>
          </h2>
        </div>
      </div>
    </div>
  );
};

export default PlayerCard;
