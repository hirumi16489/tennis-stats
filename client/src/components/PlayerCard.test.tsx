import renderer from 'react-test-renderer';
import PlayerCard from './PlayerCard';

const player = {
  id: 52,
  firstname: 'Novak',
  lastname: 'Djokovic',
  shortname: 'N.DJO',
  sex: 'M',
  country: {
    picture: 'https://i.eurosport.com/_iss_/geo/country/flag/medium/6944.png',
    code: 'SRB',
  },
  picture: 'https://i.eurosport.com/_iss_/person/pp_clubteam/large/565920.jpg',
  data: {
    rank: 2,
    points: 2542,
    weight: 80000,
    height: 188,
    age: 31,
    last: [1, 1, 1, 1, 1],
  },
};

it('renders correctly', () => {
  const tree = renderer.create(<PlayerCard player={player} />).toJSON();
  expect(tree).toMatchSnapshot();
});
