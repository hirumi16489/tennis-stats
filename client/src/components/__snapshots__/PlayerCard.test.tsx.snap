// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="banner"
    style={
      Object {
        "backgroundImage": "url(https://i.eurosport.com/_iss_/geo/country/flag/medium/6944.png)",
      }
    }
  >
    <img
      alt="Avatar"
      src="https://i.eurosport.com/_iss_/person/pp_clubteam/large/565920.jpg"
    />
  </div>
  <div
    className="menu"
  >
    <div
      className="opener"
    >
      <span />
      <span />
      <span />
    </div>
  </div>
  <h2
    className="name"
  >
    Novak
     
    Djokovic
  </h2>
  <div
    className="title"
  >
    N.DJO
  </div>
  <div
    className="info-wrapper"
  >
    <div
      className="info"
    >
      <h2>
        <div>
          <span>
            2
          </span>
          <small>
            Rank
          </small>
        </div>
      </h2>
      <h2>
        <div>
          <span>
            2542
          </span>
          <small>
            Points
          </small>
        </div>
      </h2>
    </div>
    <div
      className="info"
    >
      <h2>
        <div>
          <span>
            80
          </span>
          <small>
            Weight
          </small>
        </div>
      </h2>
      <h2>
        <div>
          <span>
            1.88
          </span>
          <small>
            Height
          </small>
        </div>
      </h2>
    </div>
    <div
      className="info single"
    >
      <h2>
        <div>
          <span>
            31
          </span>
          <small>
            Age
          </small>
        </div>
      </h2>
    </div>
  </div>
</div>
`;
